### PR TITLE
tests: test on different browsers

### PR DESCRIPTION
**Pull request by naterush**
# Description

1. Tests on Linux: Chromium, Firefox, Chrome
2. Tests on Mac: Chromium, Firefox, Webkit, Chrome
3. Tests on Windows: Chromium, Firefox, Chrome, Edge

# Testing

See new tests!

# Documentation

See updated README.md

### CODE REVIEW

File: tests/streamlit_ui_tests/features/merge.spec.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ test.describe('Merge', () => {
     });
 
 
-    test('Replays Dependent Edits', async ({ page }) => {
+    test.skip('Replays Dependent Edits', async ({ page }) => {
```
Comment (aarondr77) [inline tests/streamlit_ui_tests/features/merge.spec.ts:40]: Is this not passing on one of the browsers?

File: tests/streamlit_ui_tests/streamlit.spec.ts [hunk 2]
```diff
@@ -645,7 +650,11 @@ test.describe('Keyboard Shortcuts', () => {
     await expect(mito.locator('.index-header-selected').nth(1)).toHaveText('2');
   })
 
-  test('Select All', async ({ page }) => {
+  test.skip('Select All', async ({ page, browserName }) => {
+    if (browserName === 'webkit') {
```
Comment (aarondr77) [inline tests/streamlit_ui_tests/streamlit.spec.ts:654]: Webkit is Safari right? Did these tests catch the issue that these features aren't working on safari? cool!

Let's open an issue for them if so

File: .github/workflows/test-mitosheet-frontend.yml [hunk 3]
```diff
@@ -56,18 +62,24 @@ jobs:
...
 
   test-mitosheet-frontend-streamlit:
-    runs-on: ubuntu-20.04
     timeout-minutes: 60
     strategy:
       matrix:
+        os: ['ubuntu-latest', 'macos-latest', 'windows-latest']
         python-version: ['3.10']
-        run_count: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+        project: ['chromium', 'firefox', 'webkit', 'Google Chrome', 'Microsoft Edge']
```
Comment (aarondr77) [inline .github/workflows/test-mitosheet-frontend.yml:70]: Looks like we are only testing on Python 3.6, 3.8, 3.10, maybe a next step for testing should be testing on all versions of Python we purport to support. Python 3.11 included

**Review by aarondr77 — COMMENTED** [review]
This is awesome! Setting up testing for all of these environments is huge.

File: tests/streamlit_ui_tests/features/merge.spec.ts (hunk 1 above)
Comment (naterush) [reply tests/streamlit_ui_tests/features/merge.spec.ts:40]: Not passing on any. We can fix up later.

File: tests/streamlit_ui_tests/streamlit.spec.ts (hunk 2 above)
Comment (naterush) [reply tests/streamlit_ui_tests/streamlit.spec.ts:654]: It's not really an issue, Safari just does not support these shortcuts.

**Pull request merged**